pkg/scheduler/flavorassigner/tas_flavorassigner.go

```diff
@@ -20,61 +20,69 @@ import (
 	"errors"
 	"fmt"
 
-	"github.com/go-logr/logr"
 	"k8s.io/utils/ptr"
 
 	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
 	"sigs.k8s.io/kueue/pkg/cache"
+	"sigs.k8s.io/kueue/pkg/features"
 	"sigs.k8s.io/kueue/pkg/workload"
 )
 
-func assignTopology(log logr.Logger,
-	psAssignment *PodSetAssignment,
+// TopologyRequestsFor - returns the TopologyRequests of the workload
+func (a *Assignment) TopologyRequestsFor(wl *workload.Info,
+	requests []workload.PodSetResources,
+	cq *cache.ClusterQueueSnapshot) map[kueue.ResourceFlavorReference][]cache.TASPodSetRequests {
+	var tasRequests map[kueue.ResourceFlavorReference][]cache.TASPodSetRequests
+	if features.Enabled(features.TopologyAwareScheduling) {
```

Comment:
Is this if necessary, as we already check this before calling this function?

Reply:
Not necessary, in the main preemption PR I'm calling it without the outer feature-gate check, but I will revisit this in that PR. Dropping the double check for now.